packages/react-native-layout/src/components/container/Container.tsx

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { View, ViewStyle } from 'react-native';
+
+import type { TestableComponent } from '../../types/generic-props';
+import type { ContainerProps } from './Container.props';
+
+const Container: React.FC<ContainerProps & TestableComponent> = ({
+  children,
+  testID,
+  type,
+  backgroundColor = 'transparent',
+  top = 0,
+  right = 0,
+  bottom = 0,
+  left = 0,
+  horizontal = 0,
+  vertical = 0,
+  onLayout,
+}) => {
+  const styles: ViewStyle = {
+    backgroundColor,
+    ...(horizontal
+      ? {
+          [`${type}Horizontal`]: horizontal,
+        }
+      : {
+          [`${type}Left`]: left,
+          [`${type}Right`]: right,
+        }),
+    ...(vertical
+      ? {
+          [`${type}Vertical`]: vertical,
+        }
+      : {
+          [`${type}Top`]: top,
+          [`${type}Bottom`]: bottom,
+        }),
```

Comment:
oh my god hahaha.
I feel like splitting this up here would increase readability a lot

Reply:
Don't fully agree with this.

As this is a library, and the code shouldn't be touched often, brevity trumps readability in my opinion. 🤓

Reply:
In my view, this is a bothrs library which might be used for a very long time by a lot more engineers, so readability should be the top concern, the compiler will make it shorter